### PR TITLE
Make refactoring of the project's structure

### PR DESCRIPTION
**Pull request by vasayxtx**
This PR reorganizes the project structure to enhance clarity and modularity. The key changes include:

* **Relocation of Cross-domain Typed Identifiers (CTI) Parser**: The CTI parser and associated structs have been moved to the root Go module for improved accessibility. Previously located in ./pkg/identifier, this change addresses difficulties locating these core components.
* **Creation of Dedicated Go Modules**:
  + **Metadata Module**: All other Go packages are now grouped into a separate module, github.com/acronis/go-cti/metadata, for better organization and separation of concerns.
  + **CLI Module**: The command-line tool (cti) has been isolated into its own Go module to prevent CLI-related dependencies from being included in the root go.mod, thus keeping it lightweight and focused.

These structural improvements aim to make the codebase more maintainable and user-friendly by clearly delineating core components, metadata, and CLI functionality.

**Future Plans**: In an upcoming PR, once all versions are set via Git tags, dependencies will be specified directly in each go.mod, allowing the removal of go.work for a cleaner, more streamlined dependency management setup.

### CODE REVIEW

File: doc.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+Copyright © 2024 Acronis International GmbH.
+
+Released under MIT license.
+*/
+
+// Package cti contains Cross-domain Typed Identifiers (CTI) parser and related utilities.
```
Comment (viatoriche) [inline doc.go:7]: 👍

File: metadata/collector/collector.go [hunk 2]
```diff
@@ -424,7 +423,7 @@ func (c *Collector) getCtiSchema(base *raml.BaseShape, ctiSchema *raml.DomainExt
 }
 
 func (c *Collector) readCtiCti(base *raml.BaseShape) ([]string, error) {
```
Comment (viatoriche) [inline metadata/collector/collector.go:425]:
```suggestion
func (c *Collector) readMetadataCti(base *raml.BaseShape) ([]string, error) {
```

File: metadata/collector/collector.go [hunk 3]
```diff
@@ -183,20 +182,20 @@ func (c *Collector) MakeCtiTypeFromShape(id string, shape *raml.BaseShape) (*cti
 	return entity, nil
 }
 
-func (c *Collector) MakeCtiInstanceFromExtension(id string, definedBy *raml.ArrayShape, values map[string]interface{}, valuesLocation string) *cti.Entity {
+func (c *Collector) MakeCtiInstanceFromExtension(id string, definedBy *raml.ArrayShape, values map[string]interface{}, valuesLocation string) *metadata.Entity {
```
Comment (viatoriche) [inline metadata/collector/collector.go:185]:
```suggestion
func (c *Collector) MakeMetadataInstanceFromExtension(id string, definedBy *raml.ArrayShape, values map[string]interface{}, valuesLocation string) *metadata.Entity {
```

File: metadata/collector/collector.go [hunk 4]
```diff
@@ -117,7 +116,7 @@ func (c *Collector) Collect() error {
 	return nil
 }
 
-func (c *Collector) MakeCtiTypeFromShape(id string, shape *raml.BaseShape) (*cti.Entity, error) {
+func (c *Collector) MakeCtiTypeFromShape(id string, shape *raml.BaseShape) (*metadata.Entity, error) {
```
Comment (viatoriche) [inline metadata/collector/collector.go:119]:
```suggestion
func (c *Collector) MakeMetadataTypeFromShape(id string, shape *raml.BaseShape) (*metadata.Entity, error) {
```

File: metadata/collector/collector.go [hunk 5]
```diff
@@ -6,20 +6,19 @@ import (
...
+	"github.com/acronis/go-cti/metadata"
 	"github.com/acronis/go-raml"
-
-	"github.com/acronis/go-cti/pkg/cti"
 )
 
-var annotationsToMove = []string{cti.Reference, cti.Schema}
+var annotationsToMove = []string{metadata.Reference, metadata.Schema}
 
 type CtiRegistry struct {
```
Comment (viatoriche) [inline metadata/collector/collector.go:16]:
```suggestion
type MetadataRegistry struct {
```

File: metadata/collector/unwrap.go [hunk 6]
```diff
@@ -3,9 +3,8 @@ package collector
 import (
 	"fmt"
 
+	"github.com/acronis/go-cti/metadata"
 	"github.com/acronis/go-raml"
-
-	"github.com/acronis/go-cti/pkg/cti"
 )
 
 func (c *Collector) unwrapCtiType(base *raml.BaseShape) (*raml.BaseShape, error) {
```
Comment (viatoriche) [inline metadata/collector/unwrap.go:10]:
```suggestion
func (c *Collector) unwrapMetadataType(base *raml.BaseShape) (*raml.BaseShape, error) {
```

File: metadata/validator/validator.go [hunk 7]
```diff
@@ -269,10 +268,12 @@ func (v *CtiValidator) GetMergedSchema(id string) (map[string]interface{}, error
 	return schema, nil
 }
 
-func (v *CtiValidator) FindInheritedAnnotation(id string, key cti.GJsonPath, predicate func(*cti.Annotations) bool) *cti.Annotations {
+func (v *CtiValidator) FindInheritedAnnotation(
```
Comment (viatoriche) [inline metadata/validator/validator.go:271]:
```suggestion
func (v *MetadataValidator) FindInheritedAnnotation(
```

**Review by viatoriche — APPROVED** [review]
Please fix minor items if you can

File: metadata/collector/collector.go (hunk 2 above)
Comment (vasayxtx) [reply metadata/collector/collector.go:425]: Good point, thanks. How about using UPPER case for initialisms (`readMetadataCti` -> `readMetadataCTI`)? It's a [common practice in Go](https://go.dev/wiki/CodeReviewComments#initialisms).

**Pull request merged**